druid-shell/src/platform/windows/window.rs

```diff
@@ -595,6 +613,29 @@ impl WndProc for MyWndProc {
                     let s = s.as_mut().unwrap();
                     let x = LOWORD(lparam as u32) as i16 as i32;
                     let y = HIWORD(lparam as u32) as i16 as i32;
+
+                    // When the mouse first enters the window client rect we need to register for the
+                    // WM_MOUSELEAVE event. Note that WM_MOUSEMOVE is also called even when the
+                    // window under the cursor changes without moving the mouse, for example when
+                    // our window is first opened under the mouse cursor.
+                    if !s.has_mouse_focus && is_point_in_client_rect(hwnd, x, y) {
+                        s.has_mouse_focus = true;
+                        let mut desc = TRACKMOUSEEVENT {
+                            cbSize: std::mem::size_of::<TRACKMOUSEEVENT>() as DWORD,
+                            dwFlags: TME_LEAVE,
+                            hwndTrack: hwnd,
+                            dwHoverTime: HOVER_DEFAULT,
+                        };
+                        unsafe {
+                            if TrackMouseEvent(&mut desc) == FALSE {
+                                warn!(
+                                    "failed to TrackMouseEvent: {}",
+                                    Error::Hr(HRESULT_FROM_WIN32(GetLastError()))
+                                );
```

Comment:
This is an interesting failure possibility. I think it might be better to set `s.has_mouse_focus` to `false` so that we'll try again on the next move event. Otherwise we'll forever be without this tracking.

Reply:
I'm curious in what circumstances this call would fail, besides being passed wrong arguments. But good call with not setting `has_mouse_focus`.